components/templates/src/global_fns/load_data.rs

```diff
@@ -201,6 +207,19 @@ impl TeraFn for LoadData {
         };
         let path_arg = optional_arg!(String, args.get("path"), GET_DATA_ARGUMENT_ERROR_MESSAGE);
         let url_arg = optional_arg!(String, args.get("url"), GET_DATA_ARGUMENT_ERROR_MESSAGE);
+		let post_body_arg = optional_arg!(String, args.get("body"), "`load_data` body must be a string, if set.");
+		let post_content_type = optional_arg!(String, args.get("contenttype"), "`load_data` contenttype must be a string, if set.");
+		let method_arg = optional_arg!(String, args.get("method"), "`load_data` method must either be POST or GET.");
+		let method = match method_arg {
+			Some(method_str) => {
+				if method_str.to_lowercase() == "post"{
+				Method::Post
+				} else if method_str.to_lowercase() == "get" {
+				Method::Get
+				} else {return Err("`load_data` method must either be POST or GET.".into())}
```

Comment:
Not worth it here, but there is the https://doc.rust-lang.org/std/str/trait.FromStr.html for that case in general

Reply:
implemented in the latest commit